src/components/ModalOverlay/ModalOverlay.tsx

```diff
@@ -13,10 +14,23 @@ export const ModalOverlay: React.FC<ModalOverlayProps> = ({
   onClose,
   show,
   ...rest
-}) => (
-  <Overlay className={classNames('ModalOverlay', className)} show={show} {...rest}>
-    <Modal className='ModalOverlay__modal' onClose={onClose}>
-      <div className='ModalOverlay__content'>{children}</div>
-    </Modal>
-  </Overlay>
-);
+}) => {
+  const el = useRef(document.createElement('div'));
```

Comment:
This will call `document.createElement` on every render. Can you move `createElement` into the `useEffect` hook?

Reply:
I think this is how it should be, I think that by moving it, it won't work. Can you show me? maybe I haven't understood

Reply:
btw, I think creating an element inside the useEffect would create multiple elements. this will create an element only when rendering first time 🤔

Reply:
I meant:
```
const ref = useRef();

useEffect(() => {
  ref.current = document.createElement('div');
  document.body.appendChild(ref.current);
  return () => {
    ref.current?.parentElement?.removeChild(element);
  };
}, []);

if (!ref.current) return null;

...
```

Reply:
> btw, I think creating an element inside the useEffect would create multiple elements. this will create an element only when rendering first time 🤔

It calls `document.createElement` on every render, it will initialize the ref only once but still creates an element on every render.